cache_test.go

```diff
@@ -251,3 +196,44 @@ func testGetMulti(t *testing.T, newCache cacheFactory) {
 		t.Errorf("Error getting foo: %s / %v", err, foo)
 	}
 }
+
+func testKeys(t *testing.T, newCache cacheFactory) {
+	cache := newCache(t, time.Hour)
+
+	m := map[string]interface{}{
+		"str": "foo",
+		"num": 42,
+		"foo": struct{ Bar string }{"baz"},
+	}
+
+	var keys []string
+	for key, value := range m {
+		keys = append(keys, key)
+		if err := cache.Set(key, value, time.Second*30); err != nil {
+			t.Errorf("Error setting a value: %s", err)
+		}
+	}
+
+	items, err := cache.Keys()
+	if err != nil {
+		t.Errorf("Error in Keys: %s", err)
+	}
+
+	expected := []string{"str", "num", "foo"}
+	if len(items) != len(expected) {
+		t.Errorf("Mismatching number of keys: %v != %v ", items, expected)
+	}
+
+	match := 0
+	for _, i := range expected {
```

Comment:
This piece wont work if expected = items = {2; 2}

Reply:
Keys won't duplicate themselves in a Hash to begin with.